LadybugTools_Engine/Python/src/ladybugtools_toolkit/solar.py

```diff
@@ -952,3 +914,337 @@ def plot_directional_irradiance(
+
+    # generate plot
+    rects = ax.bar(
+        x=angles,
+        height=values,
+        width=((np.pi / directions) * 2) * bar_width,
+        color=colors,
+    )
+    format_polar_plot(ax)
+
+    # add colormap
+    sm = ScalarMappable(cmap=cmap, norm=norm)
+    sm.set_array([])
+    cbar = plt.colorbar(
+        sm,
+        ax=ax,
+        orientation="vertical",
+        label="Cumulative irradiance (W/m$^2$)",
+        fraction=0.046,
+        pad=0.04,
+    )
+    cbar.outline.set_visible(False)
+
+    # add labels
+    if label:
+        offset_distance = max(values) / 10
+        if directions > 36:
+            max_angle = angles[np.argmax(values)]
+            max_val = max(values)
+            ax.text(
+                max_angle,
+                max_val + offset_distance,
+                f"{max_val:0.0f}W/m$^2$\n{np.rad2deg(max_angle):0.0f}°",
+                fontsize="xx-small",
+                ha="center",
+                va="center",
+                rotation=0,
+                rotation_mode="anchor",
+                color="k",
+            )
+        else:
+            for rect, color in list(zip(*[rects, colors])):
+                theta = rect.get_x() + (rect.get_width() / 2)
+                theta_deg = np.rad2deg(theta)
+                val = rect.get_height()
+
+                if theta_deg < 180:
+                    if val < max(values) / 2:
+                        ha = "left"
+                        anchor = val + offset_distance
+                    else:
+                        ha = "right"
+                        anchor = val - offset_distance
+                    ax.text(
+                        theta,
+                        anchor,
+                        f"{val:0.0f}",
+                        fontsize="xx-small",
+                        ha=ha,
+                        va="center",
+                        rotation=90 - theta_deg,
+                        rotation_mode="anchor",
+                        color=contrasting_color(color),
+                    )
+                else:
+                    if val < max(values) / 2:
+                        ha = "right"
+                        anchor = val + offset_distance
+                    else:
+                        ha = "left"
+                        anchor = val - offset_distance
+                    ax.text(
+                        theta,
+                        anchor,
+                        f"{val:0.0f}",
+                        fontsize="xx-small",
+                        ha=ha,
+                        va="center",
+                        rotation=-theta_deg - 90,
+                        rotation_mode="anchor",
+                        color=contrasting_color(color),
+                    )
+
+    ax.set_title(
+        f"{epw_file.name}\n{rad_type.title()} irradiance ({tilt_angle}° altitude)\n{describe_analysis_period(analysis_period)}"
+    )
+
+    plt.tight_layout()
+
+    return ax
+
+
+def tilt_orientation_factor(
```

Comment:
It does. the original Solar class was meant to be a bit more extensible to allow for creation of the radiation data from a table (as well as an EPW file), but ended up making things to complicated. I'll remove that object to save on confusion, but it would be nice to have a Solar object created from either an EPW or a pd.DataFrame (like the Wind object) to allow this kind of functionality in the future.

Will be updated in the next push.